apps/website/src/components/shared/form/MapPickerField.tsx

```diff
@@ -0,0 +1,299 @@
+import maplibregl, { GeolocateControl, Map, type Marker } from "maplibre-gl";
+import MaplibreGeocoder from "@maplibre/maplibre-gl-geocoder";
+import IconWorld from "@/icons/IconWorld";
+import IconX from "@/icons/IconX";
+import "@maplibre/maplibre-gl-geocoder/dist/maplibre-gl-geocoder.css"; // Searchbox CSS
+import "maplibre-gl/dist/maplibre-gl.css"; // Import the MapLibre CSS
+import {
+  geocoderApi,
+  getDefaultMarker,
+  reverseSearch,
+  roundCoord,
+} from "@/utils/geolocation";
+import config from "../../../../tailwind.config";
+import { CheckboxField } from "./CheckboxField";
+
+// TODO: Check for WebGL? https://maplibre.org/maplibre-gl-js/docs/examples/check-for-support/
+// TODO: when loading full data if slow then https://maplibre.org/maplibre-style-spec/layers/#icon-allow-overlap turn to true (from https://maplibre.org/maplibre-gl-js/docs/guides/large-data/#visualising-the-data)
+
+// FIXME?: some locations (like LA, San Diego or Madeira) go outside the borders when rounding coords, and they land in water next to the country or in a totally different country.
+// FIXME?: check if rounded coords are off the country and round 1 less decimal? so precission-1 and check recursively until you land on the original site? That defeats the purpose of rounding in the first place.
+
+/**
+ * @param name Unique name of the element
+ * @param initiallyHidden Initial state of the checkbox that handles the map visibility
+ * @param textToShow Text that appears next to the checkbox
+ * @param initialZoom Self describing
+ * @param allowMultipleMarkers Do we allow multiple markers in this map?
+ * @param coordsPrecission Number of decimals that we'll keep on the coordinates
+ *
+ * For the rest and additional @see {@link https://maplibre.org/maplibre-gl-js/docs/API/type-aliases/MapOptions/|Map Options}
+ */
+export type MapPickerFieldProps = {
+  name: string;
+  initiallyHidden?: boolean;
+  textToShow?: string;
+  center?: [lon: number, lat: number];
+  initialZoom?: number;
+  minZoom?: number;
+  maxZoom?: number;
+  antialias?: boolean;
+  allowMultipleMarkers?: boolean;
+  coordsPrecission?: number;
+  defaultLocation?: MapLocation;
+  onLocationChange: (userSelectedLocation: MapLocation) => void;
+};
+
+export type MapLocation = {
+  latitude: number;
+  longitude: number;
+  location: string;
+};
+
+/**
+ * Map field with integrated searchbox and geolocalization capabilities
+ *
+ * @see {@link https://maplibre.org/maplibre-gl-js/docs/API/|MapLibre API Docs}
+ * @see {@link https://maplibre.org/maplibre-gl-geocoder/types/MaplibreGeocoderApi.html|Geocoder API (Search function)}
+ * @returns
+ */
+export const MapPickerField = ({
+  defaultLocation,
+  initiallyHidden = defaultLocation?.location ? false : true,
+  textToShow = "Show map",
+  minZoom = 1,
+  maxZoom = 24,
+  initialZoom = defaultLocation?.location ? maxZoom : 1,
+  center = defaultLocation?.location
+    ? [defaultLocation.longitude, defaultLocation.latitude]
+    : [0, 0],
```

Comment:
This will create a new array on each render and since we never pass a `center` prop anyways, i would move calculating the center into the creation effect (see below).